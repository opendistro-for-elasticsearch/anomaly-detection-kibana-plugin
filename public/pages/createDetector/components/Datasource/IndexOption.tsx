/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

//@ts-ignore
import { EuiHealth, EuiHighlight } from '@elastic/eui';
import React from 'react';
import { customSuccessColor, customWarningColor, customDangerColor, customSubduedColor } from '../../../utils/constants';

type IndexOptionProps = {
  option: any;
  searchValue: string;
  contentClassName: string;
};

const healthToColor = {
  green: customSuccessColor,
  yellow: customWarningColor,
  red: customDangerColor,
  undefined: customSubduedColor,
} as { [key: string]: string };

function IndexOption({
  option,
  searchValue,
  contentClassName,
}: IndexOptionProps) {
  const { health, label, index } = option;
  const isAlias = !!index;

  const color = healthToColor[health];
  return (
    <EuiHealth color={color}>
      <span className={contentClassName}>
        <EuiHighlight search={searchValue}>{label}</EuiHighlight>
        {isAlias && (
          <span>
            &nbsp;(
            {index})
          </span>
        )}
      </span>
    </EuiHealth>
  );
}

export { IndexOption };
