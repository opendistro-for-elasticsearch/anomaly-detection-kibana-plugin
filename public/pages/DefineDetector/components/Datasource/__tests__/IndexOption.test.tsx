/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IndexOption } from '../IndexOption';

describe('<IndexOptions /> spec', () => {
  test('renders the component', () => {
    const { container } = render(
      <IndexOption
        option={{ health: 'green', label: 'logstash-1234' }}
        searchValue=""
        contentClassName=""
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders with highlight', () => {
    const { container } = render(
      <IndexOption
        option={{ health: 'green', label: 'logstash-1234' }}
        searchValue="logs"
        contentClassName=""
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
