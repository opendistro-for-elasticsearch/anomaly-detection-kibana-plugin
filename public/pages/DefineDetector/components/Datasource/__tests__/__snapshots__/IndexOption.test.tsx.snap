// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexOptions /> spec renders the component 1`] = `
<div
  class="euiHealth"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <svg
        aria-hidden="true"
        class="euiIcon euiIcon--medium euiIcon-isLoading"
        focusable="false"
        height="16"
        role="img"
        style="fill: #7DE2D1;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <span
        class=""
      >
        <span>
          logstash-1234
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`<IndexOptions /> spec renders with highlight 1`] = `
<div
  class="euiHealth"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <svg
        aria-hidden="true"
        class="euiIcon euiIcon--medium"
        focusable="false"
        height="16"
        role="img"
        style="fill: #7DE2D1;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="8"
          r="4"
        />
      </svg>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <span
        class=""
      >
        <span>
          
          <mark
            class="euiMark"
          >
            logs
          </mark>
          tash-1234
        </span>
      </span>
    </div>
  </div>
</div>
`;
